### PR TITLE
[Chat] Mention logic unit tests

### PR DESCRIPTION
**Pull request by emlynmac**
# What
Add unit tests for the mention logic handlers

# Why
Verify robustness and confidence in future changes

# How Tested
Command line `rushx test`

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+  textToTagParser,
+  updateHTML
+} from './mentionTagUtils';
+
+import { COMPONENT_LOCALE_EN_US } from '../../localization/locales';
+
+describe('Mention logic should be robust and accurate', () => {
+  const localeStrings = COMPONENT_LOCALE_EN_US.strings;
+
+  const basicMention = 'Hello <msft-mention id="1" displayText="Everyone">everyone</msft-mention>!';
```
Comment (Leah-Xia-Microsoft) [inline packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.test.ts:23]: Can the displayName be as an attr be different than the text wrapped up between the tags?

Comment (emlynmac) [reply packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.test.ts:23]: Yes, it can be edited to be that way

**Review by Leah-Xia-Microsoft — APPROVED** [review]

**Review by Leah-Xia-Microsoft — APPROVED** [review]

File: packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+    const comparisonString = 'x<y and z';
+    let parsed: { tags: TagData[]; plainText: string } | undefined;
+    expect(() => {
+      parsed = textToTagParser(comparisonString, '@');
+    }).not.toThrow();
+    expect(parsed?.plainText).toEqual(comparisonString);
+  });
+
+  test('Using > in the text does not break parsing', () => {
+    const comparisonString = 'x>y and z';
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.test.ts:123]: Should we also add a test for `/>`?

**Review by vhuseinova-msft — APPROVED** [review]

**Review by vhuseinova-msft — APPROVED** [review]

File: packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.test.ts (hunk 2 above)
Comment (emlynmac) [reply packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.test.ts:123]: We should probably add that when we get the handling of the input of < and > working

**Pull request merged**